src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -16,18 +30,74 @@
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DEPARTMENT, PREFIX_OFFICE,
+                        PREFIX_REMARK, PREFIX_TAG);
+
+        if (!areAnyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DEPARTMENT, PREFIX_OFFICE,
+                PREFIX_REMARK, PREFIX_TAG)) {
+
+            String trimmedArgs = args.trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+
+            String[] nameKeywords = trimmedArgs.split("\\s+");
+
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        }
+
+        String[] keywords;
+        List<Predicate<Person>> predicates = new ArrayList<>();
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
```

Comment:
May I also just check if there is a check for the preamble? So something like "find sjdjahdsj n/ john" would not work.

Reply:
Oops. I forgot. Thanks for reminding! 😄 Will implement it ASAP.